jax/random.py

```diff
@@ -946,6 +946,25 @@ def _gumbel(key, shape, dtype):
   return -np.log(-np.log(
       uniform(key, shape, dtype, minval=np.finfo(dtype).eps, maxval=1.)))
 
+def categorical(key, logits, axis=-1, shape=()):
```

Comment:
I see that it's easy and natural to support an `axis` argument here, but AIUI most other samplers in this file don't have an `axis` in their API, even for distributions where it might make sense, like `dirichlet` and `multivariate_normal`. (Though `shuffle` has one...)

WDYT about not having an `axis` argument in the API, to keep things simpler?

Reply:
A custom axis argument is used in the [fastar example](https://colab.research.google.com/drive/1NDvCvgqqxM0aflqzWweNtJ5P8qxrf1kr#scrollTo=me2AIAV4ildP). A workaround for the case of not having the `axis` parameter would be using `np.transpose` on `logits`. I find that the current version more readable. Not sure how common this case is though (therefore I don't have a strong opinion on this choice). 

Independently, `axis` also clarifies the behavior of `categorical` since both `-1` and `0` are conceivable defaults.

Reply:
I second this. `axis=-1` is a conventional pattern in many libraries (numpy, pytorch, tensorflow), so it would be better to have a consistent behavior here.

Reply:
SGTM, thanks for weighing in!

I'm keen to look for places where things became "conventional patterns" only because NumPy/PyTorch/TF didn't have `vmap`. Sounds like this isn't an opportunity for that though.